### ISSUE TITLE
Improve responsiveness of backend elements

### ISSUE THREAD

**Issue by ghost** [octobercms/october#4694]
Going to show two quick examples but I feel the problem extends to all the other backend elements.

### Example 1 - Buttons

I'm pretty suprised with even basic buttons in the backend are not responsive at all, see example:

![ezgif-6-7dd4903067da](https://user-images.githubusercontent.com/55155131/66845933-df176380-ef68-11e9-9aca-85c81aa3fa3a.gif)

Here we can see the search box cover the buttons. In a responsive world the buttons should be moved down a row automatically.

### Example 2 - Tables

This follows carries on from the github issue: https://github.com/octobercms/october/issues/4618

Not sure why it was closed? But there is still no solution to this issue.

In fact it gets worse when you start investigating this issue with tables and soon realize that many plugins use **partial files with the list widget** and all these tables are completely hanging out of the screen when testing October V460 (that's not been released yet).

Right now the lists widget seems to use a very old school method with css media queries and will turn on the responsive table when the screen is a certain width. Maybe the orginal design thought this would be good for mobile phones?

However, when testing tablets we have large table data hanging out of the screen and no way to scroll the data back in to view with the lists widget.

On Mobile phone devices it gets even worse and half way through the screen widths we have a css media query removing the left sidebar and then there is a huge block of screen width where you can't scroll the table at all.

Most developers will say well I use October CMS with desktop computers so who cares? Well when you create a list widget with lots of data and a few columns. I reduce the screen size down from 8K to 1920 x 1080 and the list widget data is hanging out of the screen. I only have 6 columns to see this issue. Meaning I can use the list widget for 4K and 8K computer screens and desktop screens with lower resolutuion - the data starts hanging out of the screen view and no way to pull the table data into view.

This is no good at all right now with the list widget and worse when you use a list widget in a partial file to create a secondary layer.

### Modern day solution

1. Buttons - Should not have a overflow: hidden function and get covered up.

2. Tables - Should be responsive when the data goes out of the screen view and not use a css media query to try and second guess things!

The list widget gets worse when a developer adds more columns in the table, because the list widget table will hang out of the screen at a higher screen resolution causing a bigger gap of no functionability at all.

The list widget has major issues when you start creating a second layer of data in a partial or more deeper layers. The responsive feature just goes out of the window at that point.

Basically the current setup makes a lot of features totally useless in most screen sizes.

### Out of the box

I can adjust my plugins, but these kind of features should be out of the box and applied to evereything in the backend as standard.

**Comment by Samuell1** [comment]
You see that arrow on right side ? It means you can drag scroll buttons on mobile.

**Comment by ayumihamasaki2019** [comment]
@Samuell1 

> You see that arrow on right side ? It means you can drag scroll buttons on mobile.

Yah it shows when the screen size hits a certain screen width and not when the table is hanging out of the screen.

**Comment by ayumihamasaki2019** [comment]
Possible Solution: https://github.com/filamentgroup/tablesaw

**Comment by ayumihamasaki2019** [comment]
Possible button fix:

modules/system/assets/ui/storm.css 
Code line: 3076

Old:

```css
.control-toolbar .toolbar-item .btn, .control-toolbar .toolbar-item .btn-group, .control-toolbar .toolbar-item .dropdown {
    white-space: nowrap;
    float: none;
    display: inline-block;
    margin-right: 10px;
}
```

New:

```css
.control-toolbar .toolbar-item .btn, .control-toolbar .toolbar-item .btn-group, .control-toolbar .toolbar-item .dropdown {
    white-space: nowrap;
    float: left;
    display: block;
    margin-right: 10px;
    overflow: visible;
    width: fit-content;
    margin: 0 10px 10px 0;
}
```

**Disclaime**r: Probably will throw out some error's here and there. Haven't tested it through out the cms, just wrote down some code from top of my head.

Yah as I thought issue would be old browsers not understanding modern css, e.g. https://www.caniuse.com/#feat=mdn-css_properties_width_fit-content

**Comment by Samuell1** [comment]
@ayumihamasaki2019 I told you you can drag buttons there is no need to add anything its already responsive.

**Comment by ayumihamasaki2019** [comment]
@Samuell1 

>I told you you can drag buttons there is no need to add anything its already responsive.

Sorry my bad, I thought you were talking about the table not the buttons.

p.s. What you saying doesn't work my side though ???

I just decided to inject some flexbox into my plugins to over write things for now:

![ezgif-6-30806131a679](https://user-images.githubusercontent.com/55155131/66871048-bb6d1100-ef9a-11e9-91c1-2dd502158310.gif)

Above is a list widget - I just removed the table below in the dom as the plugin is not finished yet.

As for the HTML table, it's kind of bugging me. Want to spend some time creating a nice solution that we can roll out in October to make table's better responsive and touch screen dragable.

**Comment by drmzio** [comment]
I think October could use a new UI. I'm thinking about bringing TailwindCSS to the backend and replacing the old Bootstrap + table layout css hacks. Simply using Flexbox would solve many of these issues. On a side note, also using Vuejs to replace the monster of the jQuery library.

**Comment by bennothommo** [comment]
@drmzio There's been quite a few people that have suggested doing the same. The issue with Flexbox is that it's not quite supported fully by IE11, which we still do support. It also has been reported in the past that flexbox introduces performance issues, although that may not be as much of a concern now as it was back then.

If you wanted to make a proof of concept though, you could take a look at https://octobercms.com/plugin/cyd293-backendskin which provides a way to define backend customisations in an October CMS theme.

**Comment by ayumihamasaki2019** [comment]
I don't want to get involved in this conversation 😆

- Tailwinds uses SASS and October uses LESS - so means a complete rewrite, this was my reason for creating a post on making October version 2.

- IE10 has been dropped beginning of 2020 and IE11 is still being used by Windows, bennothommo has given me the info many times.

- Flexbox has several ongoing bugs with IE11, however most issues are because many dev's are not using the correct css prefixes which correct 98% of all issues in IE.

- One way to sort out IE in October could dump a ton of polyfills in the dom for people still using IE. I'm not talking about modernizr but I was thinking more of polyfills.io

Lastly, I don't want to submit a button fix right now because I think it would require a deep test and I worry about breaking some people's plugins. But I'm happy to try and sort something out with these stupid HTML tables and make nested tables repsonsive. Looking at people's fixes on the internet they all suck and I really see responsive sliders as the answer.

If anyone feels they want to submit fixes to make any elements in the backend responsive, please go ahead.

**Comment by Samuell1** [comment]
This is how buttons panel works on mobile. @ayumihamasaki2019

![2019-10-16_18-22-38](https://user-images.githubusercontent.com/3110002/66938698-f4ac8c00-f041-11e9-9dae-1ebd0645e886.gif)

**Comment by bennothommo** [comment]
@amdad the fatal flaw in your argument is assuming that just developers use the backend of all installations of October.

**Comment by ayumihamasaki2019** [comment]
@Samuell1 Thanks for gif will have to dig deeper and try to find out why that's not working my side, thanks for the info 👍

**Comment by ayumihamasaki2019** [comment]
@LukeTowers Done some research on @Samuell1 info, got the same issue as the table header with the cursor set to pointer and not grab.

Adding the following CSS:

```css
.toolbar-item.toolbar-primary.scroll-before button, .toolbar-item.toolbar-primary.scroll-before a {
    cursor: grab !important;
}
```

Makes it easier to understand and changes the styling on the top buttons and also the top navigation buttons.

Let me know if you are happy for me to create a PR fix for this?

**Comment by amdad** [comment]
> @amdad the fatal flaw in your argument is assuming that just developers use the backend of all installations of October.

Sorry, but I didn't said this at all. You need to read again. I even add scenarios to cover all human users. Even those who have still zombie browser. IE11 is not so old. It's just dead. But as ayauminagasaki said, with proper treatment it is not so broken as many would think. Even it so, there are Polyfills and other tricks are for that matter! What I was said it's only matter of time, user will need to switch to modern browsers anyway. Because Microsoft itself will kill own bastard as they did before with IE6 IE7 and so on. Remember when whole internet wants to kill that basilisk cockatrice? No one today don't even remember this. And not one sane person would say today. Hey, we need to support this IE7 thing. IE is DEAD. As Netscape, as Opera browser (when they switch to Chromium and drop Presto engine) etc. No one cares about products when even authors saying: **Don't touch it, it's dead. dead as dead. end of story.**   

No one ask me if I want to throw away my computer. Windows just update itself and hibernate my old machine forever (needed to reset bios each time to run system) because even if it was still good hardware. It was too old for Windows to drive it.  No one cares that micro percentage of users that doesn't want to update. If you care, do your work. Do not stop 98% of people from going further because someone is fan of trash metal.

**Comment by amdad** [comment]
> Tailwinds uses SASS and October uses LESS - so means a complete rewrite, this was my reason for creating a post on making October version 2.

# 😆 🤣

**Comment by ayumihamasaki2019** [comment]
@LukeTowers @bennothommo 

On the subject of IE11, could we patch this browser with all the correct - REQUIRED - polyfills ?

To make life easier, I have created the custom file for review here: https://github.com/ayumihamasaki2019/ie11-polyfill/blob/master/october-ie11-polyfill.js

Then we could load the script only for IE11 browsers using the following loader code:

```html
<script type="text/javascript">
    if(/MSIE \d|Trident.*rv:/.test(navigator.userAgent))
        document.write('<script src="october-ie11-polyfill.js"><\/script>');
</script>
```

What's your thoughts on this?
